cmd/bancheck/config/config_test.go

```diff
@@ -0,0 +1,214 @@
+								Justification: "My justification",
+								AllowedDir:    "subdirs/vetted/...",
+							},
+						},
+					},
+				},
+				Functions: []BannedFunction{},
+			},
+		},
+		{
+			desc: "file with banned function",
+			files: map[string]string{
+				"file.json": `
+				{
+					"functions": [{
+						"name": "safehttp.NewServeMuxConfig",
+						"msg": "Sample message",
+						"exemptions": [{
+							"justification": "My justification",
+							"allowedDir": "subdirs/vetted/..."
+						}]
+					}]
+				}
+				`,
+			},
+			config: &Config{
+				Imports: []BannedImport{},
+				Functions: []BannedFunction{
+					{
+						Name: "safehttp.NewServeMuxConfig",
+						Msg:  "Sample message",
+						Exemptions: []Exemption{
+							{
+								Justification: "My justification",
+								AllowedDir:    "subdirs/vetted/...",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "multiple files",
+			files: map[string]string{
+				"file1.json": `
+				{
+					"functions": [{
+						"name": "function1"
+					}],
+					"imports": [{
+						"name": "import1"
+					}]
+				}
+				`,
+				"file2.json": `
+				{
+					"functions": [{
+						"name": "function2"
+					}],
+					"imports": [{
+						"name": "import2"
+					}]
+				}
+				`,
+			},
+			config: &Config{
+				Imports: []BannedImport{
+					{Name: "import1"},
+					{Name: "import2"},
+				},
+				Functions: []BannedFunction{
+					{Name: "function1"},
+					{Name: "function2"},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			dir, cleanup, err := analysistest.WriteFiles(test.files)
+			if err != nil {
+				t.Fatalf("Test %s: WriteFiles() returned err: %v", test.desc, err)
+			}
+			defer cleanup()
+			files := make([]string, 0)
+			for f := range test.files {
+				path := filepath.Join(dir, "src", f)
+				files = append(files, path)
+			}
+
+			config, error := Read(files)
+
+			if error != nil {
+				t.Errorf("Read() got err: %v want: nil", error)
+			}
+			if diff := cmp.Diff(config, test.config); diff != "" {
+				t.Errorf("config mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+func TestConfigErrors(t *testing.T) {
+	tests := []struct {
+		desc     string            // describes the test case
+		files    map[string]string // fake workspace files
+		fileName string            // file name to read
+	}{
+		{
+			desc:     "file does not exist",
+			files:    map[string]string{},
+			fileName: "nonexistent",
+		},
+		{
+			desc:     "file is a directory",
+			files:    map[string]string{},
+			fileName: "",
+		},
+		{
+			desc: "file has invalid contents",
+			files: map[string]string{
+				"file.json": `
+				{"imports":"this should be an object"}
+				`,
+			},
+			fileName: "file.json",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			dir, cleanup, err := analysistest.WriteFiles(test.files)
+			if err != nil {
+				t.Fatalf("Test %s: WriteFiles() returned err: %v", test.desc, err)
```

Comment:
nit: `got` instead of `returned`

The standard way of printing errors at Google is (go/go-style/decisions#useful-test-failures): `"Function() got %v, want %v"`

Reply:
Changed